In [5]:
%pip install pandas

Defaulting to user installation because normal site-packages is not writeable
   ---------------------------------------- 0.0/11.1 MB ? eta -:--:--
   ----------------- ---------------------- 5.0/11.1 MB 30.2 MB/s eta 0:00:01
   --------------------------------- ------ 9.4/11.1 MB 24.2 MB/s eta 0:00:01
   ---------------------------------------- 11.1/11.1 MB 22.3 MB/s  0:00:00
   ---------------------------------------- 0.0/12.9 MB ? eta -:--:--
   -------------- ------------------------- 4.7/12.9 MB 22.7 MB/s eta 0:00:01
   -------------------------- ------------- 8.7/12.9 MB 21.2 MB/s eta 0:00:01
   ---------------------------------------  12.8/12.9 MB 20.5 MB/s eta 0:00:01
   ---------------------------------------- 12.9/12.9 MB 19.6 MB/s  0:00:00

   ---------------------------------------- 0/4 [pytz]
   ---------- ----------------------------- 1/4 [tzdata]
   ---------- ----------------------------- 1/4 [tzdata]
   -------------------- ------------------- 2/4 [numpy]
   ----------


[notice] A new release of pip is available: 25.2 -> 25.3
[notice] To update, run: python.exe -m pip install --upgrade pip


In [6]:
import pandas as pd

In [7]:
import numpy as np

Step 1 uploading data

In [17]:
df1 = pd.read_csv(
    r"C:\Users\varle\Downloads\Python heackathon Data\schema_1_ontario_final.csv"
)

df1.head()

week  fsa probable vulnerable is_most_recent fever_chills_shakes cough  \
0    13  L4C        n          n              y                   n     n   
1    13  P1H        y          y              y                   n     n   
2    14  N2J        n          y              y                   n     n   
3    13  N1G        n          n              y                   n     n   
4    13  M1P        n          y              n                   n     n   

  shortness_of_breath over_60 any_medical_conditions travel_outside_canada  \
0                   n       n                      n                     n   
1                   n       y                      y                     n   
2                   n       y                      n                     n   
3                   n       n                      n                     n   
4                   n       n                      y                     n   

  contact_with_illness  
0                    n  
1                    y  
2                    n  
3                    n  
4                    n

step 2. First look

In [19]:
df1.head()
df1.tail()
df1.shape
df1.columns
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 263639 entries, 0 to 263638
Data columns (total 12 columns):
 #   Column                  Non-Null Count   Dtype 
---  ------                  --------------   ----- 
 0   week                    263639 non-null  int64 
 1   fsa                     263639 non-null  object
 2   probable                263639 non-null  object
 3   vulnerable              263639 non-null  object
 4   is_most_recent          263639 non-null  object
 5   fever_chills_shakes     263639 non-null  object
 6   cough                   263639 non-null  object
 7   shortness_of_breath     263639 non-null  object
 8   over_60                 263639 non-null  object
 9   any_medical_conditions  263639 non-null  object
 10  travel_outside_canada   263639 non-null  object
 11  contact_with_illness    263639 non-null  object
dtypes: int64(1), object(11)
memory usage: 24.1+ MB


Insights you get here

1.Number of rows & columns

2.Column names & spelling issues

3.Data types (object, int, float, datetime)

4.Missing values

Step 3 Understand each column’s meaning

Create a mental or written mapping:

Column	Meaning 	    Expected Type
date	record date	    datetime
week	time period	    int
cases	count	        int
region	location	   category

 Step 4:DATA QUALITY CHECKS

 4️ Missing values analysis

In [21]:
df1.isna().sum().sort_values(ascending=False)

week                      0
fsa                       0
probable                  0
vulnerable                0
is_most_recent            0
fever_chills_shakes       0
cough                     0
shortness_of_breath       0
over_60                   0
any_medical_conditions    0
travel_outside_canada     0
contact_with_illness      0
dtype: int64

Strategy:

Numerical → mean / median / 0

Categorical → mode / "Unknown"

Critical column → drop rows

Step 5  Duplicate records

In [22]:
df1.duplicated().sum()

np.int64(241441)

👉 241,441 rows in our dataset are duplicates

Why this matters (VERY IMPORTANT)

1.Duplicate rows inflate counts

2.Trends become misleading

3.Aggregations (sum, mean) become wrong

What you should do next
1️⃣ Check total rows

In [23]:
df1.shape

(263639, 12)

2.Inspect duplicate rows

In [24]:
df1[df1.duplicated()].head()

week  fsa probable vulnerable is_most_recent fever_chills_shakes cough  \
86     13  M4L        n          n              y                   n     n   
91     13  L1M        n          y              n                   n     n   
94     14  L7E        n          n              y                   n     n   
107    13  L3C        n          n              y                   n     n   
125    13  L4C        n          n              y                   n     n   

    shortness_of_breath over_60 any_medical_conditions travel_outside_canada  \
86                    n       n                      n                     n   
91                    n       y                      n                     y   
94                    n       n                      n                     n   
107                   n       n                      n                     n   
125                   n       n                      n                     n   

    contact_with_illness  
86                     n  
91                     n  
94                     n  
107                    n  
125                    n

In [27]:
df1_clean = df1.drop_duplicates()

3. Verify duplicates are removed

In [28]:
df1_clean.duplicated().sum()

np.int64(0)

This confirms duplicates are handled correctly.

What our duplicate preview tells us
1.Rows look identical across all columns
2.Same week, FSA, symptoms, flags

**Correct Data Cleaning Order
Reset index

In [29]:
df1_clean.reset_index(drop=True, inplace=True)

Why we are reset index

After dropping rows, index becomes messy

Clean index helps analysis & merges

In [ ]:
Understand our columns

In [30]:
df1_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22198 entries, 0 to 22197
Data columns (total 12 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   week                    22198 non-null  int64 
 1   fsa                     22198 non-null  object
 2   probable                22198 non-null  object
 3   vulnerable              22198 non-null  object
 4   is_most_recent          22198 non-null  object
 5   fever_chills_shakes     22198 non-null  object
 6   cough                   22198 non-null  object
 7   shortness_of_breath     22198 non-null  object
 8   over_60                 22198 non-null  object
 9   any_medical_conditions  22198 non-null  object
 10  travel_outside_canada   22198 non-null  object
 11  contact_with_illness    22198 non-null  object
dtypes: int64(1), object(11)
memory usage: 2.0+ MB


 we Look for:

object columns (strings like y/n)

numeric columns (week)

missing values (null count)

Standardize binary columns.
Our data has many yes/no fields.

Convert them to 0/1 (clean & analytical):

In [31]:
binary_cols = [
    'probable', 'vulnerable', 'is_most_recent',
    'fever_chills_shakes', 'cough',
    'shortness_of_breath', 'over_60',
    'any_medical_conditions', 'travel_outside_canada'
]

df1_clean[binary_cols] = df1_clean[binary_cols].replace({'y': 1, 'n': 0})

C:\Users\varle\AppData\Local\Temp\ipykernel_13564\3047509522.py:8: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df1_clean[binary_cols] = df1_clean[binary_cols].replace({'y': 1, 'n': 0})
C:\Users\varle\AppData\Local\Temp\ipykernel_13564\3047509522.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1_clean[binary_cols] = df1_clean[binary_cols].replace({'y': 1, 'n': 0})


This enables:

correlations

ML models

clean aggregations

Check missing values

In [32]:
df1_clean.isna().sum().sort_values(ascending=False)

week                      0
fsa                       0
probable                  0
vulnerable                0
is_most_recent            0
fever_chills_shakes       0
cough                     0
shortness_of_breath       0
over_60                   0
any_medical_conditions    0
travel_outside_canada     0
contact_with_illness      0
dtype: int64

Create INSIGHT-Weekly case count,Symptom prevalence

In [34]:
df1_clean[binary_cols].mean().sort_values(ascending=False)

is_most_recent            0.760699
vulnerable                0.565637
any_medical_conditions    0.396747
probable                  0.366204
over_60                   0.330030
cough                     0.243220
travel_outside_canada     0.194882
shortness_of_breath       0.185017
fever_chills_shakes       0.132670
dtype: float64

In [33]:
weekly_cases = df1_clean.groupby('week').size()

Download Clean Data

In [35]:
df1_clean.to_csv(
    "clean_schema_1_ontario.csv",
    index=False
)

In [37]:
df1_clean.to_csv(
    r"C:\Users\varle\Downloads\clean_schema_1_ontario.csv",
    index=False
)